### PR TITLE
Add User-Agent header to all API calls

### PR DESCRIPTION
**Pull request by Didainius**
This PR will use new govcd functionality to send HTTP User-Agent (introduced in https://github.com/vmware/go-vcloud-director/pull/336).
The sample header `User-Agent: [terraform-provider-vcd/v2.9.0-16-gb1cfc7b5 (darwin/amd64; isProvider:true)]` (for stable version - `User-Agent: [terraform-provider-vcd/v2.9.0 (darwin/amd64; isProvider:true)]`)
Version is to be output of [`git describe`](https://git-scm.com/docs/git-describe). OS is described by GOOS and GOARCH. `isProvider` is set if `Sysorg==System`.

`git describe` is injected in during build time using `go build -ldflags` flag inside Makefile. 
**Note** If user uses `go install` the `BuildVersion` variable will remain default ("unset")

In addition this PR bumps Travis config to use Go 1.15 for testing as this is the version used now.

### CODE REVIEW

**Review by vbauzysvmware — COMMENTED** [review]
Firs pass. LGTM
`Added support for VCD 10.2` - what was change to make it work?

File: vcd/config.go [hunk 1]
```diff
@@ -424,10 +432,15 @@ func (c *Config) Client() (*VCDClient, error) {
 		return nil, fmt.Errorf("something went wrong while retrieving URL: %s", err)
 	}
 
+	userAgent := fmt.Sprintf("terraform-provider-vcd/%s (%s/%s; isProvider:%t)",
+		BuildVersion, runtime.GOOS, runtime.GOARCH, strings.ToLower(c.SysOrg) == "system")
```
Comment (lvirbalas) [inline vcd/config.go:436]: I see the User-Agent with the new build:

Old build (grep):
```
2020/10/02 09:59:45 	Vary: [Accept-Encoding, User-Agent]
2020/10/02 09:59:46 	Vary: [Accept-Encoding, User-Agent]
...
```

New build (grep):
```
2020/10/05 11:52:30 	User-Agent: [terraform-provider-vcd/v2.9.0-28-g5b5d93d (darwin/amd64; isProvider:true)]
2020/10/05 11:52:31 	Vary: [Accept-Encoding, User-Agent]
2020/10/05 11:52:31 	User-Agent: [terraform-provider-vcd/v2.9.0-28-g5b5d93d (darwin/amd64; isProvider:true)]
2020/10/05 11:52:32 	User-Agent: [terraform-provider-vcd/v2.9.0-28-g5b5d93d (darwin/amd64; isProvider:true)]
2020/10/05 11:52:32 	User-Agent: [terraform-provider-vcd/v2.9.0-28-g5b5d93d (darwin/amd64; isProvider:true)]
2020/10/05 11:52:33 	Vary: [Accept-Encoding, User-Agent]
2020/10/05 11:52:33 	User-Agent: [terraform-provider-vcd/v2.9.0-28-g5b5d93d (darwin/amd64; isProvider:true)]
2020/10/05 11:52:33 	Vary: [Accept-Encoding, User-Agent]
...
```
Looks great in the request:
```
2020/10/05 11:52:30 Req header:
2020/10/05 11:52:30     User-Agent: [terraform-provider-vcd/v2.9.0-28-g5b5d93d (darwin/amd64; isProvider:true)]
```
Is it expected that in the response we have it empty?
```
2020/10/05 11:52:31 Response header:
2020/10/05 11:52:31     X-Vmware-Vcloud-Request-Execution-Time: [49]
2020/10/05 11:52:31     Cache-Control: [no-store, must-revalidate]
2020/10/05 11:52:31     Vary: [Accept-Encoding, User-Agent]
```

Comment (Didainius) [reply vcd/config.go:436]: > Is it expected that in the response we have it empty?

Yes. That is server sending data back (server is not expected to have/send User-Agent)

**Review by lvirbalas — APPROVED** [review]

**Comment by Didainius** [comment]
> `Added support for VCD 10.2` - what was change to make it work?

It was a change done previous by @dataclouder in go-vcloud-director (https://github.com/vmware/go-vcloud-director/pull/327) however we have not included that note in terraform-provider-vcd changelog and I think it is quite important.

**Comment by vbauzysvmware** [comment]
> > `Added support for VCD 10.2` - what was change to make it work?
> 
> It was a change done previous by @dataclouder in go-vcloud-director ([vmware/go-vcloud-director#327](https://github.com/vmware/go-vcloud-director/pull/327)) however we have not included that note in terraform-provider-vcd changelog and I think it is quite important.

Let's not forget to run full test before release then for this specific version

**Review by vbauzysvmware — APPROVED** [review]

**Review by dataclouder — APPROVED** [review]

**Pull request merged**